src/Core/InstanceConfigTrait.php

```diff
@@ -171,15 +164,38 @@ public function getConfigOrFail(string $key): mixed
      * @return $this
      */
     public function configShallow(array|string $key, mixed $value = null)
+    {
+        $this->initCfg();
+        $this->_configWrite($key, $value, 'shallow');
+
+        return $this;
+    }
+
+    /**
+     * Deletes a config key.
+     *
+     * @param string $key Key to delete. It can be a dot separated string to delete nested keys.
+     * @return $this
+     */
+    public function deleteConfig(string $key)
+    {
+        $this->initCfg();
+        $this->_configDelete($key);
+
+        return $this;
+    }
+
+    /**
+     * Initializes the config with the default config.
+     *
+     * @return void
+     */
+    protected function initCfg(): void
```

Comment:
Why use Cfg instead of Config?

Reply:
To avoid potential naming conflict is user land classes. Since this a trait it could cause problems if a class using it already has method of the same name.

Reply:
I'm learning towards we should just conflict instead of using a different naming pattern.